src/lib/actions/FocusTrap/focusTrap.ts

```diff
@@ -21,36 +20,47 @@ export function focusTrap(node: HTMLElement, enabled: boolean) {
 		}
 	}
 
-	const onInit = () => {
+	const onScanElements = (fromObserver: boolean) => {
 		if (enabled === false) return;
 		// Gather all focusable elements
 		const focusableElems: HTMLElement[] = Array.from(node.querySelectorAll(elemWhitelist));
 		if (focusableElems.length) {
 			// Set first/last focusable elements
 			elemFirst = focusableElems[0];
 			elemLast = focusableElems[focusableElems.length - 1];
-			// Auto-focus first focusable element
-			elemFirst.focus();
+			// Auto-focus first focusable element only when not called from observer
+			if (!fromObserver) elemFirst.focus();
 			// Listen for keydown on first & last element
 			elemFirst.addEventListener('keydown', onFirstElemKeydown);
 			elemLast.addEventListener('keydown', onLastElemKeydown);
 		}
 	};
-	onInit();
+	onScanElements(false);
 
-	function onDestroy(): void {
+	function onCleanUp(): void {
 		if (elemFirst) elemFirst.removeEventListener('keydown', onFirstElemKeydown);
 		if (elemLast) elemLast.removeEventListener('keydown', onLastElemKeydown);
 	}
 
+	// When children of node are changed (added or removed)
+	const onObservationChange = (mutationRecords: MutationRecord[], observer: MutationObserver) => {
+		if (mutationRecords.length) {
+			onCleanUp();
+			onScanElements(true);
+		}
+		return observer;
+	};
+	const observer = new MutationObserver(onObservationChange);
+	observer.observe(node, { childList: true, subtree: true });
```

Comment:
looking good, but I think we need to add an observer.disconnect() in the ``onCleanup()``

Reply:
you are absolutely correct, I forgot about it. will add it immediately.